crates/relayer/src/util/multihop.rs

```diff
@@ -0,0 +1,90 @@
+use ibc_relayer_types::core::ics03_connection::connection::IdentifiedConnectionEnd;
+use ibc_relayer_types::core::ics24_host::identifier::{ChainId, ConnectionId};
+pub use ibc_relayer_types::core::ics33_multihop::channel_path::{ConnectionHop, ConnectionHops};
+
+use crate::chain::handle::ChainHandle;
+use crate::chain::requests::{
+    IncludeProof, QueryClientStateRequest, QueryConnectionRequest, QueryHeight,
+};
+use crate::error::Error;
+use crate::registry::get_global_registry;
+
+pub fn build_hops_from_connection_ids(
+    src_chain_id: ChainId,
+    connection_ids: &[ConnectionId],
+) -> Result<ConnectionHops, Error> {
+    let registry = get_global_registry();
+
+    let mut connection_hops = Vec::new();
+
+    let src_connection_id = connection_ids
+        .first()
+        .ok_or(Error::empty_connection_hop_ids())?
+        .clone();
+
+    let src_chain = match registry.get_or_spawn(&src_chain_id) {
+        Ok(chain) => chain,
+        Err(_) => return Err(Error::spawn_error(src_chain_id.clone())),
+    };
```

Comment:
This can be removed since you can borrow the ChainHandle from the calling method. See other comments